libsolidity/ast/Types.cpp

```diff
@@ -865,32 +865,20 @@ bool RationalNumberType::isImplicitlyConvertibleTo(Type const& _convertTo) const
 		IntegerType const& targetType = dynamic_cast<IntegerType const&>(_convertTo);
 		unsigned forSignBit = (targetType.isSigned() ? 1 : 0);
 		if (m_value > rational(0))
-		{
-			if (m_value.numerator() <= (u256(-1) >> (256 - targetType.numBits() + forSignBit)))
-				return true;
-		}
-		else if (targetType.isSigned() && -m_value.numerator() <= (u256(1) << (targetType.numBits() - forSignBit)))
-			return true;
-		return false;
+			return m_value.numerator() <= (u256(-1) >> (256 - targetType.numBits() + forSignBit)) ? true : false;
+		if (targetType.isSigned())
+			return -m_value.numerator() <= (u256(1) << (targetType.numBits() - forSignBit)) ? true : false;
 	}
-	else if (_convertTo.category() == Category::FixedPoint)
+	if (_convertTo.category() == Category::FixedPoint)
 	{
 		if (auto fixed = fixedPointType())
 			return fixed->isImplicitlyConvertibleTo(_convertTo);
-		else
-			return false;
+		return false;
 	}
-	else if (_convertTo.category() == Category::FixedBytes)
+	if (_convertTo.category() == Category::FixedBytes)
 	{
 		FixedBytesType const& fixedBytes = dynamic_cast<FixedBytesType const&>(_convertTo);
-		if (!isFractional())
-		{
-			if (integerType())
-				return fixedBytes.numBytes() * 8 >= integerType()->numBits();
-			return false;
-		}
-		else
-			return false;
+		return !isFractional() && integerType() ? fixedBytes.numBytes() * 8 >= integerType()->numBits() : false;
```

Comment:
I actually find the previous version much better to read. It could be improved by removing the `!`, i.e.
```
if (isFractional())
  return false;
else if (integerType())
  return fixedBytes.numBytes() * 8 >= integerType()->numBits();
else
  return false;
```